### PR TITLE
fix: add missing features for Vertical Navigation 

### PR DESCRIPTION
**Pull request by InnaAtanasova**
## Related Issue
Closes SAP/fundamental-styles#2442

## Description
- changed the cursor to pointer for navigable items
- added modifier class `fd-vertical-nav__group-header` for the `fd-list__group-header` class 
- added example with Grouping
- applied `!important` to the rules changing the color of the navigated/indicated items. Couldn't rely on specificity as it was already used once. 
- In the example added another item with text and icon that is navigated to illustrate the changing of the color.

### CODE REVIEW

File: src/_listNavigationItem.scss [hunk 1]
```diff
@@ -123,12 +124,12 @@ $fd-list-second-level-popover-position: 3.25rem !default;
       .#{$block}__navigation-item-text,
       .#{$block}__navigation-item-icon,
       .#{$block}__navigation-item-arrow {
-        color: var(--sapSelectedColor);
+        color: var(--sapSelectedColor) !important;
```
Comment (InnaAtanasova) [inline src/_listNavigationItem.scss:127]: Note: there's already specificity so I couldn't rely on it, had to use !important

File: src/_listNavigationItem.scss [hunk 2]
```diff
@@ -123,12 +124,12 @@ $fd-list-second-level-popover-position: 3.25rem !default;
...
       }
     }
 
     .#{$block}__navigation-item--indicated .#{$block}__navigation-item-text {
-      color: var(--sapSelectedColor);
+      color: var(--sapSelectedColor) !important;
```
Comment (InnaAtanasova) [inline src/_listNavigationItem.scss:132]: Same note as above

File: src/vertical-nav.scss [hunk 3]
```diff
@@ -16,27 +16,15 @@ $block: #{$fd-namespace}-vertical-nav;
...
     @include fd-reset();
   }
 
   &--condensed {
     width: 3.25rem;
   }
+
+  .#{$block}__group-header {
+    background: var(--sapList_AlternatingBackground);
+  }
```
Comment (InnaAtanasova) [inline src/vertical-nav.scss:29]: removed the other code as I used for base `.fd-list__group-header` and just added this class to change the background.

File: src/_listNavigationItem.scss (hunk 1 above)
Comment (N1XUS) [reply src/_listNavigationItem.scss:127]: I wouldn't rely on `important`. Maybe add more level to it to make it prioritized if possible?

Comment (mikerodonnell89) [reply src/_listNavigationItem.scss:127]: I too am wondering why not just add another class

Comment (InnaAtanasova) [reply src/_listNavigationItem.scss]: @N1XUS I wouldn't rely on 4-5 levels of nesting to enforce specificity either :) It becomes a hell if refactoring is needed or if a new functionality is added. 
@mikerodonnell89 `fd-list__navigation-item--indicated` is the additional class added to change the color of the items. We can't have 2 classes doing the same to enforce specificity. 

Anyways, I added additional nesting. It's not a big deal, just something I don't consider a good practice :)

File: src/_listNavigationItem.scss (hunk 2 above)
Comment (InnaAtanasova) [reply src/_listNavigationItem.scss]: Refactored to rely on specificity.

**Review by N1XUS — APPROVED** [review]

**Pull request merged**